erpnext/selling/page/point_of_sale/point_of_sale.js

```diff
@@ -471,7 +471,44 @@ erpnext.pos.PointOfSale = class PointOfSale {
 				}
 			}
 
-			frappe.prompt(this.get_prompt_fields(),
+
+			let me = this;
+
+			var dialog = frappe.prompt([{
+					fieldtype: 'Link',
+					label: __('Company'),
+					options: 'Company',
+					fieldname: 'company',
+					default: me.frm.doc.company,
+					reqd: 1,
+					onchange: function(e) {
+							me.get_default_pos_profile(this.value).then((r) => {
+								if (r && r.name) {
+									dialog.set_value('pos_profile', r.name);
```

Comment:
Make existing pos_profile blank if no pos profile found against the newly selected company